.github/workflows/golang.yml

```diff
@@ -0,0 +1,64 @@
+name: golang
+# for golang basic checks with go command line tool
+on:
+  workflow_call:
+
+env:
+  GO_VERSION: "1.18"
+
+permissions:
+  contents: read
+
+jobs:
+  gofmt_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{env.GO_VERSION}}
```

Comment:
This can make use of the go-version from go.mod file see:
* https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file 
* https://github.com/sustainable-computing-io/kepler-operator/blob/v1alpha1/.github/workflows/pr-checks.yaml#L13
*